### DISCUSSION TITLE
Unable to hide facet header labels when facet is used as an encoding channel

### DISCUSSION THREAD

**Discussion by armsp** [vega/altair#3733] [General]
I have been trying to make a facet chart look like a regular bar chart (making spacing b/w facets = 0). I think it may be because the size of **labels** do not let that happen. So I tried to hide the labels to bring the facets closer but turns out hiding labels when using facet as an encoding does not seem to work -

NOTE: Using a `step` size of 4 actually gives a larger space in the first chart than the rest.

Preparation
```py
year = [2000, 2000, 2000, 2000, 2001, 2001, 2001, 2001]
quarter = [1,2,3,4,1,2,3,4]
stock = [10,2,30,-5,5,10,30,40]
d = pd.DataFrame({'year': year, 'quarter': quarter, 'stock': stock})
```
I plot this data using 4 different ways -

(Those **year** values on top of facets are the **labels**. And strangely doing `header=alt.Header(labels=False, title=None)` does not remove them. However, other properties like `labelFontSize` seems to work.)

Facet as an encoding channel fails -
-----------------------------------------------
**NOTE**: Experiment with different `step` sizes, you'd notice that if you increase step size, then the spacing b/w facets does become 0 when individual charts become larger than the label width. However, at low `step` sizes that's not the case and that's what I am pointing at cause using low step size is imperative and thus the need to hide the labels.
```py
alt.Chart(d).mark_bar().encode(
    x='quarter:O',
    y='stock:Q',
    facet=alt.Facet('year:O', bounds='flush', spacing={'column':0}, title=None, header=alt.Header(labels=False, title=None)),
).configure_axis(grid=False).configure_view(strokeWidth=1, step=4)
```
![2_1](https://user-images.githubusercontent.com/12681262/89004326-89594f80-d31f-11ea-83ef-b3793e7b7e38.png)

When specified via **column** channel it works -
----------------------------------------------------------------
**NOTICE**: How the spacing is much less compared to above (it looks to me that its because of the absence of labels)
```
alt.Chart(d).mark_bar().encode(
    x='quarter:O',
    y='stock:Q',
    column=alt.Column('year:O', spacing=0, title=None, header=alt.Header(labels=False)),
).configure_axis(grid=False).configure_view(strokeWidth=1, step=4)
```
![2_2](https://user-images.githubusercontent.com/12681262/89004280-6dee4480-d31f-11ea-9866-d76fa0252088.png)

Also works when specified via **column inside facet**  -
--------------------------------------------------------------------------
```py
alt.Chart(d).mark_bar().encode(
    x='quarter:O',
    y='stock:Q',
).facet(column=alt.Column('year:O', title=None, header=alt.Header(labels=False)), spacing=0).configure_axis(grid=False).configure_view(strokeWidth=1, step=4)
```
![2_3](https://user-images.githubusercontent.com/12681262/89004216-4ac39500-d31f-11ea-9a4c-b71ccd398611.png)

**Comment by jakevdp** [comment]
I apologize – I read your question three times and I'm still not certain what you're asking. The title of the question says you want to hide the facet title, but your examples seem to be mostly about setting facet spacing? In a sentence or two, what is it that you want to do, and what problem are you having in accomplishing it?

**Comment by jakevdp** [comment]
So your ultimate goal is to have zero spacing between facets, regardless of what happens to titles or labels?

**Comment by armsp** [comment]
@jakevdp any explanation why `labels=False` works for `column` but not in `facet` channel directly?

**Comment by armsp** [comment]
@jakevdp I cleaned all the unnecessary comments and I believe the issue is clearer now. Try to ignore your own replies as I have sanitized everything here. Be great if you could look into it cause it does look like an unusual behavior.

**Comment by simlmx** [comment]
I have the same issue where:
```python
alt.Facet(..., headers=alt.Headers(labels=False))
```
does not work, even though those work fine:

```python
alt.Row(..., headers=alt.Headers(labels=False))
alt.Columns(..., headers=alt.Headers(labels=False))
```


My workaround has been to use
```python
alt.Facet(..., headers=alt.Headers(labelFontSize=0))
```

**Reply by thomascamminady** [reply]
Thanks, this helped! Minor change I had to make:
Remove the `s` in `Headers` and `headers` to get

```python
alt.Facet(..., header=alt.Header(labelFontSize=0))
````
